### PR TITLE
default use ir to generate tf/xgboost code to run

### PR DESCRIPTION
**Pull request by typhoonzero**
- Default use IR codegen to generate local Tensorflow/XGBoost training code.
- Remove legacy codegen.

### CODE REVIEW

File: pkg/sql/codegen/attribute/checker.go [hunk 1]
```diff
@@ -100,6 +100,28 @@ func IntRangeChecker(lower, upper *int, includeLower, includeUpper bool) func(in
 	return checker
 }
 
+// BoolIntChecker returns a checker function that do **not** check the input.
```
Comment (shendiaomo) [inline pkg/sql/codegen/attribute/checker.go]: The comment is not inconsistent with the following code, BoolIntChecker **do** check the input.

File: pkg/sql/codegen/tensorflow/codegen.go [hunk 2]
```diff
@@ -39,6 +40,9 @@ range: [0,Infinity]`, attribute.IntRangeChecker(newInt(0), nil, false, false)},
 	"train.epoch": {attribute.Int, `[default=1]
 Number of epochs the training will run.
 range: [1, Infinity]`, attribute.IntRangeChecker(newInt(0), nil, false, false)},
+	"train.verbose": {attribute.Int, `[default=0]
+Show verbose logs when training.
+possible values: 0, 1`, attribute.IntChoicesChecker([]int{0, 1})},
```
Comment (weiguoz) [inline pkg/sql/codegen/tensorflow/codegen.go:37]: IntChoiceChecker returns an error message like `fmt.Errorf("attribute value out of range(%v), actual: %d", choices, i)`, but it's not clear for the user to find out the relative attribute name.

Comment (shendiaomo) [reply pkg/sql/codegen/tensorflow/codegen.go:37]: > IntChoiceChecker returns an error message like `fmt.Errorf("attribute value out of range(%v), actual: %d", choices, i)`, but it's not clear for the user to find out the relative attribute name.

This is an existing problem in other checkers, maybe I can fix this later in another commit

Comment (typhoonzero) [reply pkg/sql/codegen/tensorflow/codegen.go:37]: @weiguoz I'll try polish the error message in the next PR.

**Review by Yancey0623 — APPROVED** [review]
LGTM, also please @Echo9573 and @BlackPoint-CX  try to use this refactor version. And please tell us if any questions.

**Pull request merged**